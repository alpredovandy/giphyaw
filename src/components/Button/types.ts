export interface ButtonProps {
    title: string;
    onAction: () => void;
    disabled?: boolean;
    isLoading: boolean;
}
