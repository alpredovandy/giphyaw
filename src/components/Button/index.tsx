import { FC } from 'react';

import { ButtonProps } from '@/components/Button/types.ts';

const Button: FC<ButtonProps> = ({ title, onAction, disabled = false, isLoading }) => (
    <button disabled={isLoading || disabled} onClick={onAction} className="btn btn-active btn-primary items-center text-lg">
        {isLoading ? <span className="loading loading-spinner loading-sm"></span> : null} {title}
    </button>
);

export default Button;
