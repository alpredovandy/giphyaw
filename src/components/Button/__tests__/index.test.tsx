import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import Button from '@/components/Button';
import type { ButtonProps } from '@/components/Button/types';

describe('Test components: Button', () => {
    let props: ButtonProps;
    const renderButton = () => render(<Button {...props} />);

    it('Snapshot', () => {
        const { container } = renderButton();

        expect(container).toMatchSnapshot();
    });
});
