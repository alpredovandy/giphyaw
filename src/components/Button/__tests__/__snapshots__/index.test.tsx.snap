// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test components: Button > Snapshot 1`] = `
<div>
  <button
    class="btn btn-active btn-primary items-center text-lg"
  >
     
  </button>
</div>
`;
